src/h5web/visualizations/line/hooks.ts

```diff
@@ -0,0 +1,24 @@
+import { useContext, useMemo } from 'react';
+import { LineVisProps, LineVisContext } from './LineVisProvider';
+import { Domain } from '../shared/models';
+import { extendDomain, findDomain } from '../shared/utils';
+
+export function useProps(): LineVisProps {
+  const props = useContext(LineVisContext);
+
+  if (!props) {
+    throw new Error('Missing LineVis provider.');
+  }
+
+  return props;
+}
+
+export function useDomains(): [Domain, Domain | undefined] {
+  const { data } = useProps();
+  const dataDomain = useMemo(() => findDomain(data), [data]);
+
+  return [
+    extendDomain([0, data.length - 1], 0.05),
+    dataDomain ? extendDomain(dataDomain, 0.05) : undefined,
+  ];
```

Comment:
You could just return an `AxisDomains` object so you could pass it directly to `VisCanvas`. I think it would make sense to return `undefined` for the whole object, as there's not much we can do with just the abscissa domain. This would also be consistent with mu suggestion above to make `axisDomain` optional in `VisCanvas`.